internal/bgp/frr/config.go

```diff
@@ -102,6 +102,9 @@ router bgp {{$r.MyASN}}
 {{- if ne .BFDProfile ""}} 
   neighbor {{.Addr}} bfd profile {{.BFDProfile}}
 {{- end }}
+{{- if and (eq .IPFamily.String "ipv6") (ne $r.MyASN .ASN) }}
```

Comment:
How about adding a custom "mustDisableConnectedCheck" with a comment of the reason on its definition instead of the and-ed if?

See https://github.com/metallb/metallb/blob/889a4bded1b46a19d5c99053f6440781439ff00e/internal/bgp/frr/config.go#L225